### PR TITLE
planner, executor: set memory size to MAXUint64 for prepare plan cache test

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### What problem does this PR solve? 
Set memory size to MAXUint64 for prepare plan cache test

### What is changed and how it works?
The test for prepare plan cache is unstable caused by the machine memory utilization after #8339 .
So I set the memory size to MAX_Uint64 to make the test result stable every time.

### Check List 

Tests 

 - No code

Code changes
N/A

Side effects
N/A

Related changes
N/A

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8556)

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
PTAL @eurekaka @winoros

File: executor/prepared_test.go [hunk 1]
```diff
@@ -47,7 +46,9 @@ func (s *testSuite) TestPrepared(c *C) {
 		plannercore.SetPreparedPlanCache(flag)
 		plannercore.PreparedPlanCacheCapacity = 100
 		plannercore.PreparedPlanCacheMemoryGuardRatio = 0.1
-		plannercore.PreparedPlanCacheMaxMemory, err = memory.MemTotal()
+		// PreparedPlanCacheMaxMemory is set to MAX_UINT64 to make sure the cache
+		// behavior would not be effected by the uncertain memory utilization.
+		plannercore.PreparedPlanCacheMaxMemory = math.MaxUint64
 		c.Assert(err, IsNil)
```
Comment (dbjoa) [inline executor/prepared_test.go]: Would you remove the line of the error checking, which is related with ``memory.MemTotal()``?

File: executor/prepared_test.go [hunk 2]
```diff
@@ -253,7 +254,9 @@ func (s *testSuite) TestPreparedLimitOffset(c *C) {
...
```
Comment (dbjoa) [inline executor/prepared_test.go]: ditto

File: executor/prepared_test.go [hunk 3]
```diff
@@ -296,7 +299,9 @@ func (s *testSuite) TestPreparedNullParam(c *C) {
...
```
Comment (dbjoa) [inline executor/prepared_test.go]: ditto

File: executor/prepared_test.go [hunk 4]
```diff
@@ -368,7 +373,9 @@ func (s *testSuite) TestPrepareWithAggregation(c *C) {
...
```
Comment (dbjoa) [inline executor/prepared_test.go]: ditto

File: executor/prepared_test.go [hunk 5]
```diff
@@ -413,7 +420,9 @@ func (s *testSuite) TestPreparedIssue7579(c *C) {
...
```
Comment (dbjoa) [inline executor/prepared_test.go]: ditto

File: executor/prepared_test.go [hunk 6]
```diff
@@ -467,7 +476,9 @@ func (s *testSuite) TestPreparedInsert(c *C) {
...
```
Comment (dbjoa) [inline executor/prepared_test.go]: ditto

File: executor/prepared_test.go [hunk 7]
```diff
@@ -549,7 +560,9 @@ func (s *testSuite) TestPreparedUpdate(c *C) {
...
```
Comment (dbjoa) [inline executor/prepared_test.go]: ditto

File: executor/prepared_test.go [hunk 8]
```diff
@@ -608,7 +621,9 @@ func (s *testSuite) TestPreparedDelete(c *C) {
...
```
Comment (dbjoa) [inline executor/prepared_test.go]: ditto

File: executor/prepared_test.go [hunk 9]
```diff
@@ -662,7 +677,9 @@ func (s *testSuite) TestPrepareDealloc(c *C) {
 	plannercore.SetPreparedPlanCache(true)
 	plannercore.PreparedPlanCacheCapacity = 3
 	plannercore.PreparedPlanCacheMemoryGuardRatio = 0.1
-	plannercore.PreparedPlanCacheMaxMemory, err = memory.MemTotal()
+	// PreparedPlanCacheMaxMemory is set to MAX_UINT64 to make sure the cache
+	// behavior would not be effected by the uncertain memory utilization.
+	plannercore.PreparedPlanCacheMaxMemory = math.MaxUint64
 	c.Assert(err, IsNil)
```
Comment (dbjoa) [inline executor/prepared_test.go]: ditto

File: planner/core/point_get_plan_test.go [hunk 10]
```diff
@@ -48,7 +49,9 @@ func (s *testPointGetSuite) TestPointGetPlanCache(c *C) {
 	core.SetPreparedPlanCache(true)
 	core.PreparedPlanCacheCapacity = 100
 	core.PreparedPlanCacheMemoryGuardRatio = 0.1
-	core.PreparedPlanCacheMaxMemory, err = memory.MemTotal()
+	// PreparedPlanCacheMaxMemory is set to MAX_UINT64 to make sure the cache
+	// behavior would not be effected by the uncertain memory utilization.
+	core.PreparedPlanCacheMaxMemory = math.MaxUint64
 	c.Assert(err, IsNil)
```
Comment (dbjoa) [inline planner/core/point_get_plan_test.go]: ditto

File: planner/core/prepare_test.go [hunk 11]
```diff
@@ -56,7 +56,9 @@ func (s *testPrepareSuite) TestPrepareCache(c *C) {
 	core.SetPreparedPlanCache(true)
 	core.PreparedPlanCacheCapacity = 100
 	core.PreparedPlanCacheMemoryGuardRatio = 0.1
-	core.PreparedPlanCacheMaxMemory, err = memory.MemTotal()
+	// PreparedPlanCacheMaxMemory is set to MAX_UINT64 to make sure the cache
+	// behavior would not be effected by the uncertain memory utilization.
+	core.PreparedPlanCacheMaxMemory = math.MaxUint64
 	c.Assert(err, IsNil)
```
Comment (dbjoa) [inline planner/core/prepare_test.go]: ditto

File: planner/core/prepare_test.go [hunk 12]
```diff
@@ -105,7 +107,9 @@ func (s *testPrepareSuite) TestPrepareCacheIndexScan(c *C) {
...
```
Comment (dbjoa) [inline planner/core/prepare_test.go]: ditto

File: planner/core/prepare_test.go [hunk 13]
```diff
@@ -138,7 +142,9 @@ func (s *testPlanSuite) TestPrepareCacheDeferredFunction(c *C) {
...
```
Comment (dbjoa) [inline planner/core/prepare_test.go]: ditto

File: planner/core/prepare_test.go [hunk 14]
```diff
@@ -198,7 +204,9 @@ func (s *testPrepareSuite) TestPrepareCacheNow(c *C) {
...
```
Comment (dbjoa) [inline planner/core/prepare_test.go]: ditto

**Comment by XuHuaiyu** [comment]
PTAL @dbjoa @winoros

**Comment by dbjoa** [comment]
LGTM

**Comment by dbjoa** [comment]
Please fix CI errors.

**Comment by XuHuaiyu** [comment]
PTAL @winoros @zz-jason

**Comment by zz-jason** [comment]
CI failed:
```
executor/prepared_test.go:710:49: undefined: memory
```

**Pull request merged**